docs/src/guides/Parameters.jl

```diff
@@ -0,0 +1,88 @@
+# # Handling parameters
+
+# This guide shows how to overwrite the default free parameters values and
+# how to dispatch between different parameterization options based on parameters types.
+
+# ## Overwriting parameters
+
+# `CloudMicrophysics.jl` is designed to allow easy parameter calibrations.
+# As a result, free parameters are not hard-coded in the source code but are insead
+# passed as arguments to functions. The default values are stored in a separate
+# repository [CLIMAParameters.jl](https://github.com/CliMA/CLIMAParameters.jl) in a `toml` file.
+
+# We start by importing the `CLIMAParameters` package and the needed
+# `CloudMicrophysics.jl` modules. We define the precision type.
+
+import CLIMAParameters as CP
+import CloudMicrophysics.Parameters as CMP
+import CloudMicrophysics.Microphysics1M as CM1
+import CloudMicrophysics.Microphysics2M as CM2
+import CloudMicrophysics.HetIceNucleation as CMI_het
+
+FT = Float32
+
+nothing #hide
+
+# We define the default parameters struct using the default `toml` file via the constructor
+# provided in the `CloudMicrophysics.jl` `Parameters` module.
+# Additionally, we create a `toml` file in which we save the name, value and type
+# of the parameter we are changing. In a typical application, this step would be done
+# via the calibration algorithm searching for optimal parameters values.
+# We create a second struct with changed parameters based on the `override_dict.toml` file.
+# Finally we check the values of the rain formation timescales and the corresponding
+# rain formation rates.
+const default = CMP.Rain(FT)
+
+override_file = joinpath("override_dict.toml")
+open(override_file, "w") do io
+    println(io, "[rain_autoconversion_timescale]")
+    println(io, "value = " * string(1))
+    println(io, "type = \"float\"")
+end
+toml_dict = CP.create_toml_dict(FT; override_file)
+isfile(override_file) && rm(override_file; force=true)
+const overwrite = CMP.Rain(FT, toml_dict)
+
+qₗ = FT(1e-3)
+default_acnv = CM1.conv_q_liq_to_q_rai(default.acnv1M, qₗ) # Rain autoconversion rate
+overwrite_acnv = CM1.conv_q_liq_to_q_rai(overwrite.acnv1M, qₗ) # Rain autoconversion rate
+
+@info("Default:", default.acnv1M.τ, default_acnv)
+@info("Overwrite:", overwrite.acnv1M.τ, overwrite_acnv)
+
+# ## Dispatching over parameter types
+
+# `CloudMicrophysics.jl` `Parameters` module introduces type hierarchy that is used to
+# dispatch over different parameterization options and inputs.
+# For example `dust_activated_number_fraction` accepts `ArizonaTestDust` or `DesertDust`
```

Comment:
This is great except I'm about to get rid of this function. Maybe `ABIFM_J` with Illite or Kaolinite is a better function to show as an example (specifically not saying desert dust because I can't find where we got those parameters in the Knopf&Alpert 2013 paper rn)?

Reply:
```
const aerosol1 = CMP.Illite(FT)
const aerosol2 = CMP.Kaolinite(FT)

Δa_w = 0.28
J1 = CMI_het.ABIFM_J(aerosol1, Δa_w)
J2 = CMI_het.ABIFM_J(aerosol2, Δa_w)

@info("ABIFM derived J: ", J1, " and ", J2)
```